### PR TITLE
fix(app): [KDL6-112] handle automount and secret for existing service account

### PR DESCRIPTION
**Pull request by igzjaviergil**
handle automount and secret for existing service account

## Motivation and Context

Existing service account prior to upgrade to kubernetes 1.26 were created without automount. 
It means, on start up syncronize service account must check if existing service account has automount. If not, set to true and create corresponding Secret Type ServiceAccountToken
https://intelygenz.atlassian.net/browse/KDL6-112

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] I have created tests for my code changes, and the tests are passing.
- [X] I have executed the pre-commit hooks locally.
- [ ] I have updated the documentation accordingly.
- [ ] The commit message follows our guidelines: https://github.com/konstellation-io/kdl-server/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Service accounts created prior to migration to kubernetes 1.26 dont have autmount set to true and corresponding Secret Type ServiceAccountToken. With this change, if service account doesn´t have autmount to true, service account will be updated and secret created.

Closes #

## Does this PR introduce a breaking change?

- [] Yes
- [X] No

### CODE REVIEW

File: app/api/infrastructure/k8s/service_account.go [hunk 1]
```diff
@@ -72,13 +89,54 @@ func (k *K8sClient) CreateUserServiceAccount(ctx context.Context, usernameSlug s
...
+		err = k.createSecretTypeServiceAccountToken(ctx, saSecretName, usernameSlug)
+
+		if err != nil {
+			return nil, err
+		}
+
+		k.logger.Info("The service account was created in k8s correctly", "name", serviceAccount.Name)
+
+		return serviceAccount, nil
+	}
```
Comment (alexrodfe) [inline app/api/infrastructure/k8s/service_account.go:128]: ey, cuidado aqui falta una condicion
else if err != nil && !k8errors.IsNotFound(err) return err

File: app/api/infrastructure/k8s/service_account_test.go [hunk 2]
```diff
@@ -82,14 +89,47 @@ func (s *serviceAccountTestSuite) TearDownTest() {
...
 
-	sa, err := s.client.GetUserServiceAccount(context.Background(), "sa")
+	sa, err := s.client.GetUserServiceAccount(context.Background(), saSimpleName)
+	s.NoError(err)
+
+	s.Equal(*sa.AutomountServiceAccountToken, true)
+	s.Equal(saSecretName, sa.Secrets[0].Name)
+
+	_, err = s.client.GetSecret(context.Background(), sa.Secrets[0].Name)
 	s.NoError(err)
```
Comment (alexrodfe) [inline app/api/infrastructure/k8s/service_account_test.go:105]: These should be s.require().NoError as an error is blocking in this pipeline

File: app/api/infrastructure/k8s/service_account_test.go [hunk 3]
```diff
@@ -82,14 +89,47 @@ func (s *serviceAccountTestSuite) TearDownTest() {
...
+	// Arrange, create a service account without automount
+	saNoAutomount, err := s.clientset.CoreV1().ServiceAccounts(namespace).Create(context.Background(), &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: saName,
+		},
+	}, metav1.CreateOptions{})
+	s.NoError(err)
+	s.NotNil(saNoAutomount)
+	s.Nil(saNoAutomount.AutomountServiceAccountToken)
+	s.Equal(len(saNoAutomount.Secrets), 0)
```
Comment (alexrodfe) [inline app/api/infrastructure/k8s/service_account_test.go:118]: you can s.Len here better

**Review by alexrodfe — CHANGES_REQUESTED** [review]
Genial!
Te dejo algunas cosas que he visto

File: app/api/infrastructure/k8s/service_account_test.go (hunk 3 above)
Comment (igzjaviergil) [reply app/api/infrastructure/k8s/service_account_test.go:118]: what is the difference?

File: app/api/infrastructure/k8s/service_account.go (hunk 1 above)
Comment (igzjaviergil) [reply app/api/infrastructure/k8s/service_account.go:128]: en donde exactamente falta?

**Review by alexrodfe — APPROVED** [review]

**Pull request merged**